docs/docs/how-to/cloud/working-with-redirects-and-rewrites.md

```diff
@@ -0,0 +1,223 @@
+  conditions: {
+    country: `it`
+  }
+})
+
+createRedirect({
+  fromPath: `/blog`,
+  toPath: `/english/blog`,
+  conditions: {
+    country: [`us`, `gb`]
+  }
+})
+```
+
+You can also return status's 404,403,451,500 if you would like to have non redirect pages returned.
+The example below will return a 404 page for all users located in the United States (us).
+
+```js:title=gatsby-node.js
+createRedirect({
+  fromPath: `/*`,
+  toPath: `/`,
+  statusCode: 404,
+  conditions: {
+    country: `us`
+  }
+})
+```
+
+The next example will return 451 page users located in the United States (us) and Canada (ca) from all paths. Essentially, all hits from US and CA will be sent to a 451.
+
+```js:title=gatsby-node.js
+createRedirect({
+  fromPath: `/*`,
+  toPath: `/`,
+  statusCode: 451,
+  conditions: {
+    country: [`us`, `ca`],
+  }
+})
+```
+
+If you have defined a custom status page: 404, 403, 451, 500, that custom page will be rendered and sent to your users. If you haven’t defined a custom status page, Gatsby Cloud will return a generic status page. Only if you give `createRedirect` a redirect statusCode: 3XX will it return a redirect to the user.
+
+## Query Param redirects
+
+Query params allows you to further control URL matches. For example, /param?id=7redirects to /gatsby_file.pdf
+
+```js:title=gatsby-node.js
+createRedirect({
+  fromPath: `/param?id=7`,
+  toPath: `/gatsby_file.pdf`,
+})
+```
+
+A more complex example is described here, where we will redirect from /param?id=idto /param/:id
+
+```js:title=gatsby-node.js
+createRedirect({
+  fromPath: `/param?id=:id`,
+  toPath: `/param/:id`,
+})
+```
+
+### Language based redirects
+
+Language based redirects use the Accept-Language header in a request to match a route and perform a redirect. This allows you to redirect your site to the site that will match your user's language. We use the Two-letter code identifier defined by ISO 639-1. The example below will redirect users with en Accept-Language header to /en/book_list and users with zh to /zh/book_list when they reach out to /book_list
+
+```js:title=gatsby-node.js
+createRedirect({
+  fromPath: `/book_list`,
+  toPath: `/en/book_list`,
+  conditions: {
+    language: [`en`],
+  },
+})
+
+createRedirect({
+  fromPath: `/book_list/*`,
+  toPath: `/zh/book_list/*`,
+  conditions: {
+    language: [`zh`],
+  },
+})
+```
+
+### ☝️Tip: Using a Static File to Manage Redirects
+
+You might want to simplify managing your redirects by using a static file to store redirects. Storing that file in your Gatsby project can simplify your code and make it easier fo other contributors to add more redirects.
+
+At Gatsby, we use a YAML file to store our redirects so that the whole team can easily add new redirects without having to change the gatsby-node file. You could also use a JSON file if your team is comfortable with that.
+
+Here's how you could do it for our recipes site using a JSON file:
+
+```json
+// redirects.json
```

Comment:
Comments like this are incorrect syntax for JSON. Should use the `:title=redirects.json` formatting to notate that: https://www.gatsbyjs.com/contributing/gatsby-style-guide/#code-formatting-titles